OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/consistency/impl/ConsistencyManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.onesignal.common.consistency.impl
+
+import com.onesignal.common.IConsistencyManager
+import com.onesignal.common.consistency.ICondition
+import com.onesignal.common.consistency.OffsetKey
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * Manages offsets that function as read-your-write tokens for more accurate segment membership
+ * calculation. Uses customizable conditions that block retrieval of the newest offset until met.
+ *
+ * Usage:
+ *  val consistencyManager = ConsistencyManager()
+ *  val updateConditionDeferred = consistencyManager.registerCondition(MyCustomCondition())
+ *  val newestUpdateOffset = updateConditionDeferred.await()
+ */
+class ConsistencyManager : IConsistencyManager {
```

Comment:
Add generic type to replace `OffsetKey` to decouple logic

e.g. `class ConsistencyManager<T>`